src/pybind/mgr/cephadm/service_discovery.py

```diff
@@ -231,6 +235,19 @@ def ceph_exporter_sd_config(self) -> List[Dict[str, Collection[str]]]:
             })
         return srv_entries
 
+    def nfs_exporter_sd_config(self) -> List[Dict[str, Collection[str]]]:
+        """Return <http_sd_config> compatible prometheus config for nfs-exporter service."""
+        srv_entries = []
+        for dd in self.mgr.cache.get_daemons_by_type('nfs'):
+            assert dd.hostname is not None
+            addr = dd.ip if dd.ip else self.mgr.inventory.get_addr(dd.hostname)
+            port = NFSService.DEFAULT_EXPORTER_PORT
```

Comment:
Looks like a lot of the other sections for this do `dd.ports[0] if dd.ports else <some-default-port>`. If that doesn't cause issues here I'd say stick with that setup.

Reply:
Well currently `nfs-ganehsa` exporter has default port set to `9587` & there's no way yet to customize this port value in nfs config. So for now we should use default port of nfs exporter (& not service port)

Reply:
`dd.ports[0]` is not from a config or anything, it's the port cephadm found the daemon listening on when doing its normal sweep of the daemons on the nodes. Although, now that I think about it, nfs already binds to port 2049 anyway, so maybe for now this is best. We'd have to find out which entry in the list is that port and which is the monitoring one.